tests/Definitions/CustomAttribute.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Square\Pjson\Tests\Definitions;
+
+use Square\Pjson\Json;
+use Square\Pjson\JsonSerialize;
+
+class CustomAttribute
+{
+    use JsonSerialize;
+
+    #[Json]
+    public string $key;
+
+    #[Json]
+    public int|string $value;
```

Comment:
We need some tests with `int|string|null` 

We also need to decide what to do if the type is `Class1|Class2`